assembl/static2/js/app/utils/httpRequestHandler.js

```diff
@@ -1,24 +1,46 @@
+import urljoin from 'url-join';
+import Cookies from 'js-cookie';
+
+import { getCSRFToken } from './csrf';
+import { basePath } from './server';
+
 const convertToURLEncodedString = obj =>
   Object.keys(obj)
     .map(k => `${encodeURIComponent(k)}=${encodeURIComponent(obj[k])}`)
     .join('&');
 const getResponseContentType = xhr =>
   // eslint-disable-line no-unused-vars
   xhr.getResponseHeader('Content-Type').split(';')[0];
+
+const useCSRFProtection = document.getElementById('useCSRFProtection')
+  ? document.getElementById('useCSRFProtection').value
+  : 'false';
+
 /*
   A global async method that returns a Promisified ajax call
   @params payload [Object] The object that will be sent
   @params isJson [Boolean] Pass a flag if the object is JSON. Default is form header.
   @retuns [Promise]
 */
+
 export const xmlHttpRequest = obj =>
-  new Promise((resolve, reject) => {
+  new Promise(async (resolve, reject) => {
```

Comment:
Could just remove the `new Promise` and make xmlHttpRequest an `async` func and await the result/error then

Reply:
I think that changes upstream code. For example, the code using this function is written like:

```
  return xmlHttpRequest({
    method: 'POST',
    url: route,
    isJson: true,
    payload: payload
  });
```

We would have to change it all for await xmlHttpRequest

Reply:
Okie